### PR TITLE
Overhaul of our long-term-storage docs

### PR DESCRIPTION
**Pull request by thomasvn**
## Related issue

N/A

## Proposed Changes

* Most still had references to Thanos and were incorrectly telling users to name their file `object-store.yaml`.
* Explicitly call out that federated storage is only supported in Enterprise plans
* Delete doc `aws-service-account-thanos`. Merge some of its contents about IAM policies into existing AWS long term storage doc.
* Delete section about bucket encryption because I'm unsure if that still applies.

### CODE REVIEW

**Comment by thomasvn** [comment]
@jessegoodier @srpomeroy @williamkubecost. Let me know if you find anything missing here or any glaring issues!

File: install-and-configure/install/multi-cluster/long-term-storage-configuration/long-term-storage-aws.md
Comment (williamkubecost) [inline install-and-configure/install/multi-cluster/long-term-storage-configuration/long-term-storage-aws.md:1]: One thing I noticed a lot recently on some cases is that STS Assume Role is missing permissions. I'm not sure why it seems to be the case that sometimes it is needed, and sometimes it isn't. 

```
2024-04-08T00:00:00+0000): GetCloudCost: error getting Athena columns: QueryAthenaPaginated: start query error: operation error Athena: StartQueryExecution, get identity: get credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 0459fd9b-451d-4bd0-8289-aaf90f146f37, api error AccessDenied: User: arn:aws:sts::185511261685:assumed-role/aws-prod-us-east-1-tech-initial-eks-node-group/i-05c1fa9d0eb168e35 is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::331236996304:role/KubecostRole-185511261685
2024-04-04T00:00:00+0000): GetCloudCost: error getting Athena columns: QueryAthenaPaginated: start query error: operation error Athena: StartQueryExecution, get identity: get credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 6494b54b-1a9e-47ea-941d-e316cb0bc778, api error AccessDenied: User: arn:aws:sts::185511261685:assumed-role/aws-prod-us-east-1-tech-initial-eks-node-group/i-05c1fa9d0eb168e35 is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::331236996304:role/KubecostRole-185511261685
```

But might be worth a shout out!

- https://docs.aws.amazon.com/IAM/latest/UserGuide/troubleshoot_roles.html

**Review by williamkubecost — COMMENTED** [review]
I made a comment about STS assume role, I think it may be worth mentioning. The overhaul looks great, so thank you for this!

 I'll leave a little bit of time for others to review, but BRAVO!

File: install-and-configure/install/multi-cluster/long-term-storage-configuration/long-term-storage-aws.md
Comment (thomasvn) [reply install-and-configure/install/multi-cluster/long-term-storage-configuration/long-term-storage-aws.md:1]: Really good feedback! I'll add this to a `Troubleshooting` section.

**Review by bstuder99 — APPROVED** [review]

**Comment by jessegoodier** [comment]
Don't wait on me, looks good at first glance.

**Review by bstuder99 — APPROVED** [review]

**Pull request merged**